### PR TITLE
[GLUTEN-1897][CH] Respect hdfs-site.xml configs for clickhouse libhdfs3

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?

(Fixes: \#1898)

### CODE REVIEW

**Comment by taiyang-li** [comment]
Firstly we apply 4 spark hadoop configs to clickhouse backend: 
- spark.hadoop.input.connect.timeout: default 180s
- spark.hadoop.input.write.timeout: default 180s
- spark.hadoop.input.read.timeout: default 180s
- spark.hadoop.dfs.client.log.severity: default "INFO"

Above spark.hadoop.* configs maybe from spark-submit configs or hdfs-site.xml file(the former has higher priority). If no config key is specified, default value is applied in CH libhdfs3. 

```
 --conf spark.hadoop.input.connect.timeout=600000 \
 --conf spark.hadoop.input.read.timeout=600000 \
 --conf spark.hadoop.input.write.timeout=600000 \
 --conf spark.hadoop.dfs.client.log.severity="DEBUG2" \
```

**Comment by taiyang-li** [comment]
@zzcclp do you hive time to review this pr, thanks!

**Comment by taiyang-li** [comment]
image

download files failed, can you help retrigger the failed uts @zzcclp

File: backends-clickhouse/src/main/java/io/glutenproject/vectorized/CHNativeExpressionEvaluator.java [hunk 1]
```diff
@@ -56,6 +56,7 @@ public void initNative(SparkConf conf) {
 
     // Get the customer config from SparkConf for each backend
     BackendsApiManager.getTransformerApiInstance().postProcessNativeConfig(nativeConfMap, prefix);
+    // System.out.println("Native backend config: " + nativeConfMap.toString());
```
Comment (zzcclp) [inline backends-clickhouse/src/main/java/io/glutenproject/vectorized/CHNativeExpressionEvaluator.java:59]: remove

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 2]
```diff
@@ -383,7 +383,11 @@ object GlutenConfig {
         COLUMNAR_VELOX_CONNECTOR_IO_THREADS.defaultValueString),
       (
         COLUMNAR_VELOX_SPLIT_PRELOAD_PER_DRIVER.key,
-        COLUMNAR_VELOX_SPLIT_PRELOAD_PER_DRIVER.defaultValueString)
+        COLUMNAR_VELOX_SPLIT_PRELOAD_PER_DRIVER.defaultValueString),
+      ("spark.hadoop.input.connect.timeout", "180000"),
```
Comment (zzcclp) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:387]: get the value of these config from config ?

File: backends-clickhouse/src/main/java/io/glutenproject/vectorized/CHNativeExpressionEvaluator.java (hunk 1 above)
Comment (taiyang-li) [reply backends-clickhouse/src/main/java/io/glutenproject/vectorized/CHNativeExpressionEvaluator.java:59]: I kept it for future similar debugs.

**Comment by taiyang-li** [comment]
image
 
failed velox uts seems not related to this pr.

**Comment by rui-mo** [comment]
@taiyang-li Could you rebase to the main branch? This issue should be fixed.

File: backends-clickhouse/src/main/java/io/glutenproject/vectorized/CHNativeExpressionEvaluator.java (hunk 1 above)
Comment (zzcclp) [reply backends-clickhouse/src/main/java/io/glutenproject/vectorized/CHNativeExpressionEvaluator.java:59]: dont keep such code

Comment (zzcclp) [reply backends-clickhouse/src/main/java/io/glutenproject/vectorized/CHNativeExpressionEvaluator.java:59]: maybe use logDebug to print

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by rui-mo — APPROVED** [review]

**Pull request merged**